server/channels/store/sqlstore/integrity.go

```diff
@@ -544,3 +547,61 @@ func CheckRelationalIntegrity(ss *SqlStore, results chan<- model.IntegrityCheckR
 	mlog.Info("Done with relational integrity checks")
 	close(results)
 }
+
+func GetAllValidUserIDsFromDMChannels(ss *SqlStore) ([]*model.User, error) {
+	records := []model.OrphanedRecord{}
+	dirtyIDs := []string{}
+	query, args, err := ss.getQueryBuilder().
+		Select().
+		Column("CT.ChannelName AS ParentId").
+		From("Channels AS CT").
+		Where(sq.Eq{"CT.Type": []model.ChannelType{model.ChannelTypeDirect}}).
+		ToSql()
+
+	if err != nil {
+		mlog.Error("There is an error with the DM Channel Name query")
+		return nil, err
+	}
+
+	err = ss.GetMasterX().Select(&records, query, args...)
+	if err != nil {
+		mlog.Error("there us an issue fetching the records for DM Channel Names")
+		return nil, err
+	}
+
+	for k := range records {
+		dirtyIDs = append(dirtyIDs, *records[k].ParentId)
+	}
+
+	userIDs := []string{}
+
+	for _, v := range dirtyIDs {
+		temp := strings.Split(v, "_")
+		userIDs = append(userIDs, temp[0])
```

Comment:
Why not `temp[1]`?

Reply:
At the time of writing this using `temp[0]` felt more natural to me since the string being processed would look like `apple23__apple23` given the idx 0 and idx 1 were the same.

Is it industry standard or more idiomatic to use `temp[1]`? (I genuinely appreciate any feedback I can get.)

Reply:
idx0 and idx1 should be different. Because those are different userIDs. If they are the same, then it's an error.

Reply:
I have to check who is the initiator and who is the reciever, but yeah that is the logic.